service/src/test/java/greencity/service/HabitAssignServiceImplTest.java

```diff
@@ -289,10 +289,23 @@ void unenrollHabitThrowWrongIdException() {
     @Test
     void deleteHabitAssign() {
         HabitAssign habitAssign = ModelUtils.getHabitAssign();
-        when(habitAssignRepo.findByUserIdAndHabitId(1L, 1L)).thenReturn(Optional.ofNullable(habitAssign));
+        habitAssign.getHabit().setHabitAssigns(generateHabitAssignList());
+        when(habitAssignRepo.findByUserIdAndHabitId(1L, 1L)).thenReturn(habitAssign.getHabit().getHabitAssigns());
         assert habitAssign != null;
         habitAssignService.deleteHabitAssign(1L, 1L);
-        verify(habitAssignRepo).delete(habitAssign);
+        verify(habitAssignRepo, times(2)).save(any());
+    }
+
+    private List<HabitAssign> generateHabitAssignList() {
```

Comment:
please add a test if `habitAssign status` is not equals to `HabitAssignStatus.INPROGRESS `